services/blocks/standard/lmdfinalizer/finalizer.go

```diff
@@ -0,0 +1,203 @@
+// Package lmdfinalizer establishes which blocks are finalized from its LDM votes and the LDM votes of its children blocks.
+// Abbreviation: LFB means Latest Finalized Block.
+package lmdfinalizer
+
+import (
+	"encoding/hex"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/wealdtech/chaind/services/blocks/standard/lmdfinalizer/tree"
+	"github.com/wealdtech/chaind/services/chaindb"
+)
+
+// PreBalanceThreshold is a mock threshold used until we connect the finalizer with the validator balances.
+const PreBalanceThreshold = 10000
+
+// LMDFinalizer is a beacon chain finalizer based on LMD votes.
+type LMDFinalizer interface {
+	// AddBlock to finalizer to be candidate for finalization and use its included attestations as votes for
+	// other blocks.
+	AddBlock(dbblock *chaindb.Block, attestations []*chaindb.Attestation)
+}
+
+// newLFBHandler event handler to be triggered when a new LFB is finalized.
+type newLFBHandler func(phase0.Root, phase0.Slot)
+
+// finalizer is the implementation of LMDFinalizer.
+type finalizer struct {
+	tree  tree.Tree
+	votes lmdVotes
+
+	log zerolog.Logger
+
+	onAddNode chan *tree.Node
+
+	newLFBHandler newLFBHandler
+}
+
+// New LMDFinalizer with logger `log`.
+func New(latestFinalized *chaindb.Block, log zerolog.Logger, handler newLFBHandler) LMDFinalizer {
```

Comment:
Yes, ideally everything but the context would be a parameter as it makes future changes significantly easier to manager (for example, if we wanted to allow multiple handlers we could just add a `WithHandlers()` alongside the additional `WithHandler()` and it would not break any existing code.